iOSClient/Utility/NCUtility.swift

```diff
@@ -748,19 +748,47 @@ class NCUtility: NSObject {
     }
 
     func isDirectoryE2EE(serverUrl: String, userBase: NCUserBaseUrl) -> Bool {
-        return isDirectoryE2EE(serverUrl: serverUrl, account: userBase.account, urlBase: userBase.urlBase, userId: userBase.userId)
+        return isDirectoryE2EE(account: userBase.account, urlBase: userBase.urlBase, userId: userBase.userId, serverUrl: serverUrl)
     }
+
     func isDirectoryE2EE(file: NKFile) -> Bool {
-        return isDirectoryE2EE(serverUrl: file.serverUrl, account: file.account, urlBase: file.urlBase, userId: file.userId)
+        return isDirectoryE2EE(account: file.account, urlBase: file.urlBase, userId: file.userId, serverUrl: file.serverUrl)
     }
-    @objc func isDirectoryE2EE(serverUrl: String, account: String, urlBase: String, userId: String) -> Bool {
+
+    func isDirectoryE2EE(account: String, urlBase: String, userId: String, serverUrl: String) -> Bool {
         if serverUrl == NCUtilityFileSystem.shared.getHomeServer(urlBase: urlBase, userId: userId) || serverUrl == ".." { return false }
-        if let directory = NCManageDatabase.shared.getTableDirectory(predicate: NSPredicate(format: "account == %@ AND serverUrl == %@", account, serverUrl)) {
+        if let directory = NCManageDatabase.shared.getTableDirectory(account: account, serverUrl: serverUrl) {
             return directory.e2eEncrypted
         }
         return false
     }
 
+    func isDirectoryE2EETop(account: String, serverUrl: String) -> Bool {
+        if let url = URL(string: serverUrl)?.deletingLastPathComponent() {
+            if let directory = NCManageDatabase.shared.getTableDirectory(account: account, serverUrl: String(url.absoluteString.dropLast())) {
+                return !directory.e2eEncrypted
+            }
+        }
+        return true
+    }
+
+    func getDirectoryE2EETop(serverUrl: String, account: String) -> tableDirectory? {
+        var serverUrl = serverUrl
+        var top = NCManageDatabase.shared.getTableDirectory(account: account, serverUrl: serverUrl)
+        while true {
+            if let url = URL(string: serverUrl)?.deletingLastPathComponent() {
+                serverUrl = String(url.absoluteString.dropLast())
+                if let directory = NCManageDatabase.shared.getTableDirectory(account: account, serverUrl: serverUrl) {
+                    if directory.e2eEncrypted {
+                        top = directory
+                    } else {
+                        return top
+                    }
+                } else { return top }
+            } else { return top }
+        }
+    }
```

Comment:
This looks a bit complex with a lot of elses that do the same thing. Maybe something like this would be better:

```suggestion
func getDirectoryE2EETop(serverUrl: String, account: String) -> tableDirectory? {
    var serverUrl = serverUrl
    var top: tableDirectory? = nil
    
    while let url = URL(string: serverUrl)?.deletingLastPathComponent(),
          let directory = NCManageDatabase.shared.getTableDirectory(account: account, serverUrl: serverUrl) {
        
        if directory.e2eEncrypted {
            top = directory
        } else {
            return top
        }
        
        serverUrl = String(url.absoluteString.dropLast())
    }
    
    return top
}
```